pkg/daemon/ceph/osd/daemon.go

```diff
@@ -375,6 +375,13 @@ func getAvailableDevices(context *clusterd.Context, agent *OsdAgent) (*DeviceOsd
 						logger.Infof("device %q matches device filter %q", device.Name, desiredDevice.Name)
 					}
 				} else if desiredDevice.IsDevicePathFilter {
+					if device.Type == sys.PartType {
+						device, err = clusterd.PopulateDeviceUdevInfo(device.Name, context.Executor, device)
```

Comment:
Since we need to detect this for partitions, how about moving this up inside the code block on line 308 that already checks for partitions? Then it is just a single check instead of calling the same thing in two places.

Reply:
Thanks, fixed.